### PR TITLE
Updating Cake, Cake.Recipe, Packages, and project structure.

### PR DESCRIPTION
**Pull request by phillipsj**
@gep13 is this what you were asking to be completed?

### CODE REVIEW

File: src/Cake.PinNuGetDependency/PinNuGetDependancyAliases.cs [hunk 1]
```diff
@@ -11,10 +11,20 @@
...
 {
+    /// <summary>
+    /// Contains functionality for pinning NuGet dependencies.
+    /// </summary>
     [CakeAliasCategory("NuGet")]
     [CakeNamespaceImport("Cake.PinNuGetDependency")]
     public static class PinNuGetDependencyAliases
     {
+        /// <summary>
+        ///
```
Comment (glennawatson) [inline src/Cake.PinNuGetDependency/PinNuGetDependancyAliases.cs:22]: Probably should give it a comment

**Review by glennawatson — CHANGES_REQUESTED** [review]

**Comment by gep13** [comment]
@phillipsj essentially, yes.  This project is further along the lines than some of the others that I pointed you at, but the overhaul that you have provided here does bring it into line with some of the conventions that are going across other addins within the Cake Contrib Organisation.

If @glennawatson and others have no objections, it would be great to get this merged in.

**Comment by glennawatson** [comment]
Apart from the comment it looks good to me.

**Comment by phillipsj** [comment]
Nice catch.  I know what happened, I got distracted and didn't make it
back. I will fix it tonight and squash.

**Comment by gep13** [comment]
Relates to #24

**Comment by gep13** [comment]
@phillipsj @glennawatson I am going to get this merged, and fixed up.

**Comment by phillipsj** [comment]
Thanks.  I have been so busy my list is too long and not enough time.

**Comment by gep13** [comment]
@phillipsj not a problem at all.  Working my way through my list, and this one was on it 😄

**Pull request merged**